src/analysis/souffle/tag_transforms.dl

```diff
@@ -0,0 +1,156 @@
+//-----------------------------------------------------------------------------
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-----------------------------------------------------------------------------
+#ifndef SRC_ANALYSIS_SOUFFLE_TAG_TRANSFORMS_DL_
+#define SRC_ANALYSIS_SOUFFLE_TAG_TRANSFORMS_DL_
+
+#include "src/analysis/souffle/operations.dl"
+#include "src/analysis/souffle/tags.dl"
+
+// Note: this will need to change when we integrate auth logic into the SQL
+// verifier.
+#define DEFAULT_SQL_OWNER "sql"
+#define TAG_TRANSFORM_OPERATOR "sql.tag_transform"
+#define TAG_TRANSFORM_RULE_NAME_ATTR "rule_name"
+
+.type SqlPolicyRulePrecondition = [ preconditionName : symbol, requiredTag: IntegrityTag ]
```

Comment:
Mostly a clarification: is the `preconditionName` the name `a` and `b` from a pattern such as `a? + b?`?

Reply:
Yes, that is correct.

Reply:
Added a comment explaining this.